File: MyExampleIOT/app/src/main/java/com/solie/myexampleiot/MainActivity.java
package com.solie.myexampleiot;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageButton;
import android.widget.Toast;

import com.solie.myexampleiot.helper.NetworkStatus;

public class MainActivity extends AppCompatActivity {

    ImageButton powerButton;
    boolean powerOnAndOff = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        powerButton = findViewById(R.id.imageButton);

        NetworkCheck();
    }

    public void NetworkCheck() {
        int status = NetworkStatus.getConnectivityStatus(getApplicationContext());
        if (status == NetworkStatus.TYPE_NON_CONNECTED) {
            Toast.makeText(getApplicationContext(),"인터넷에 연결되지 않았습니다. 연결을 확인해주세요.", Toast.LENGTH_SHORT).show();
            powerButton.setClickable(false);
        } else {
            powerButton.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if(!powerOnAndOff) {
                        powerOnAndOff = true;
                        powerButton.setImageResource(R.drawable.power_on_icon);
                        Toast.makeText(getApplicationContext(),"Power On", Toast.LENGTH_SHORT).show();
                    } else if (powerOnAndOff) {
                        powerOnAndOff = false;
                        powerButton.setImageResource(R.drawable.power_off_icon);
                        Toast.makeText(getApplicationContext(),"Power Off", Toast.LENGTH_SHORT).show();
                    }
                }
            });
        }
    }
}

File: EV_Nyang/app/src/main/java/com/solie/ev_nyang/MainActivity.kt
package com.solie.ev_nyang

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.solie.ev_nyang.databinding.ActivityMainBinding
import com.solie.ev_nyang.firebase.ChargerInfo
import com.solie.ev_nyang.util.FirebaseData
import com.solie.ev_nyang.util.NetworkStatus

class MainActivity : AppCompatActivity(), FirebaseData {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        networkCheck()

        binding.SetFB.setOnClickListener {
            ChargerInfo.getInfo(3)
            val intent = Intent(applicationContext, MapActivity::class.java)
            startActivity(intent)

        }

    }

    private fun networkCheck() {
        val networkStatus = NetworkStatus(applicationContext).checkNetworkStatus()
        if (!networkStatus) {
            Toast.makeText(applicationContext, "인터넷에 연결되지 않았습니다.", Toast.LENGTH_LONG).show()
            finish()
        } else {

        }
    }
}